learn/ai_powered_search/rest_response_request.mdx

```diff
@@ -0,0 +1,341 @@
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": "{{text}}"
+    },
+    "response": {
+      "data": {
+        "text": "{{embedding}}"
+      }
+    }
+  }
+}
+```
+
+Providers might return embeddings in an array:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": "{{text}}"
+    },
+    "response": {
+      "data": [
+        {
+          "embedding": "{{embedding}}"
+        },
+        "{{..}}"
+      ]
+    }
+  }
+}
+```
+
+The first item in `response.data` tells Meilisearch where to find the embedding. The second item must be `"{{..}}"`, and it indicates there are multiple embeddings in the response.
+
+Cloudflare's embeddings are instead located in `response.result.data`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "text": "{{text}}"
+    },
+    "response": {
+      "result": {
+        "data": "[{{embedding}}]"
+      }
+    }
+  }
+}
+```
+
+The prompt data type does not necessarily match the response data type. For example, Cloudflare always returns an array of embeddings, even if the prompt in your request was a string. If you know the response will always contain a single embedding, you may omit `{{..}}` from the second array item.
+
+Meilisearch silently ignores `response` fields not pointing to `{{embedding}}`.
+
+## The embedding header
+
+Your provider might also request you to add specific headers to your request. For example, Azure's AI services require a `api-key` header containing an API key.
+
+Add the `headers` field to your embedder object:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "text": "{{text}}"
+    },
+    "response": {
+      "result": {
+        "data": "[{{embedding}}]"
+      }
+    },
+    "headers": {
+      "FIELD_NAME": "FIELD_VALUE"
+    }
+  }
+}
+```
+
+By default, Meilisearch includes a `Content-Type` header. It may also include an authorization bearer token, if you have supplied an API key.
+
+## Configure remainder of the embedder
+
+`source`, `request`, `response`, and `header` are the only fields specific to REST embedders.
+
+Like other remote embedders, you're likely required to supply an `apiKey`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "url": "PROVIDER_URL",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": ["{{text}}", "{{..}}"],
+      "encoding_format": "float"
+    },
+    "response": {
+      "data": [
+        {
+          "embedding": "{{embedding}}"
+        },
+        "{{..}}"
+      ]
+    },
+    "apiKey": "PROVIDER_API_KEY",
+  }
+}
+```
+
+You should also set a `documentTemplate`. Good templates are short and include only highly relevant document data:
```

Comment:
Why? A link to documentTemplate could be cool here!

Reply:
It will come as soon as we have the document template best practices ready 😉